server/monitor-types/tls.js

```diff
@@ -0,0 +1,290 @@
+const { MonitorType } = require("./monitor-type");
+const { log, UP } = require("../../src/util");
+const net = require("net");
+const tls = require("tls");
+const unescape = require("unescape-js");
+
+/**
+ * Encloses a string in double quotes.
+ * @param {string} s String to quote
+ * @returns {string} Quoted string
+ */
+function quote(s) {
+    return JSON.stringify(s);
+}
+
```

Comment:
The reason why I went with `JSON.stringify()` is that some of the logged strings span multiple lines. This tends to mess up the log output, so I wanted to avoid that. Also, with raw logging there's always the security aspect with special escape sequences doing stuff to the terminal.

Here's an example of the multi-line output a HTTP server generates showing up in the logs:
```
2024-06-04T09:30:40+02:00 [MONITOR] DEBUG: [TLS.03: POP3 good] Next heartbeat in: 59869ms
2024-06-04T09:30:40+02:00 [PORT-TLS] DEBUG: read data: 'HTTP/1.1 200 OK
Content-Length: 6
Connection: close
Content-Type: text/plain
Date: Tue, 04 Jun 2024 07:30:39 GMT
Server: Kestrel
Set-Cookie: ARRAffinity=c5e90f0d5e6f37f5e2cc43fc8b88c4ecdb32ccbcfaec54b421e66970b3c785de;Path=/;HttpOnly;Secure;Domain=httpstat.us
Set-Cookie: ARRAffinitySameSite=c5e90f0d5e6f37f5e2cc43fc8b88c4ecdb32ccbcfaec54b421e66970b3c785de;Path=/;HttpOnly;SameSite=None;Secure;Domain=httpstat.us
Strict-Transport-Security: max-age=2592000
Request-Context: appId=cid-v1:3548b0f5-7f75-492f-82bb-b6eb0e864e53

200 OK'
2024-06-04T09:30:40+02:00 [PORT-TLS] DEBUG: TLS socket commanded to close
2024-06-04T09:30:40+02:00 [PORT-TLS] INFO: check successful: Data contains keyword
```
While it does look "nice" for human eyes, it gets really annoying when you're trying to grep a log file and end up with only the first line – or the opposite, when something in the HTTP header matches and you end up with a non-timestamped stray line.

Let me know what you think about these aspects. If you feel strongly about it, I'll change it, but I really think quoting+escaping is the Right Thing to do here.

Reply:
I don't think we need to consider https://www.youtube.com/watch?v=3T2Al3jdY38 (great talk, hilarious) as an active risk (given that the terminals should not support this anymore).

Given that we log as plain text (=not json) and the debug messages are mostly for development, I think that w are targeting humans
=> if the debug log is not very readable for machines, that is fine my me.

=> Escaping newlines for these debug logs is nice, but not strictly nesssesary.
=> I would keep it simple, but you can overrule this one ^^

Reply:
Apologies for the delay! I managed to watch that talk, it's really fun but apparently it had the opposite effect on me than it did on you. The one with the repeated characters alone seemed distracting enough that it tips me towards escaping. Also, in the context of TCP testing, it's pretty easy to run into binary responses on the server side and it's annoying to have to `reset` your terminal because something accidentally messed it up. So, since you offered, I'm going to overrule you on this one but if you or someone else down the line want to change it, that's fine by me. :-)